App.js

```diff
@@ -74,13 +91,66 @@ export default function App() {
       Retrive: (user, userToken) => {
         dispatch({ type: "LOGIN", user: user, token: userToken });
       },
+      SetNickname: async (name)=>{
+        try{
+          await AsyncStorage.setItem("nickname",name);
+         
+          dispatch({type:"SetNickname",name});
+        }
+        catch(e){
+          console.log(e);
+        }
+        
+      },
     }),
     []
   );
 
+  useEffect(()=>{
+    setTimeout(()=>{
+    let user=null;
+    let userToken=null;
+    AsyncStorage.getItem("userToken", (err, result) => {
+    console.log("Loading Screen..");
+    userToken=result;
+    if (result) {
+      AsyncStorage.getItem("user", (err, res) => {
+        
+        console.log(result);
+        user=JSON.parse(res);
+        dispatch({ type: "LOGIN", user: user, token: userToken });
+      });
+      AsyncStorage.getItem("nickname",(err,name)=>{
+        dispatch({type:"SetNickname",name});
+      })
+    } 
+    else{
+      dispatch({type:"LOGIN",user:null,token:null});
+    }
+  });
+    },1000)
```

Comment:
Try to use async/await or promises instead of setTimeout and callbacks @NeelMakadiya14

Reply:
Okay.
Did those changes.
Please Check.